assets/js/page-enhancements.js

```diff
@@ -0,0 +1,164 @@
+    $('select#prisonId').each((idx, element) => {
+      accessibleAutocomplete.enhanceSelectElement({
+        selectElement: element,
+        source: suggest,
+      })
+    })
+    $('#prisonId')
+      .closest('form')
+      .submit(event => {
+        // Before submitting the form set the prisonId select option to the value matching the autocomplete, or the null option if there is no match
+        // This is because the autocomplete currently only sets the select option when there is a valid match, which causes incorrect
+        // behaviour when the user performs a valid autocomplete, then changes it to blank or non-matching text. In this case the underlying
+        // select is not reset so we have to do it ourselves.
+        event.preventDefault()
+        const underlyingSelectField = $('#prisonId-select')
+        const autoCompleteField = $('#prisonId')
+        // In all cases start off by setting the underlying select option to the null/empty option
+        underlyingSelectField.children(`option[value='']`).prop('selected', true)
+        // Now find the option whose text matches the autocomplete field, and select it. If there is no match nothing will be selected
+        underlyingSelectField
+          .children('option')
+          .filter((idx, option) => option.text === autoCompleteField.val())
+          .prop('selected', true)
+        // Submit the form
+        event.target.submit()
+      })
+  }
+
+  const enableCopyBarcodeButton = () => {
+    if (typeof ClipboardItem !== 'function') {
+      return
+    }
+    $('.barcode-address-copy-button')
+      .css('display', 'inline-flex')
+      .click(event => {
+        const container = $(event.target).closest('div.barcode-address-container')
+        const image = container.find('img.barcode-address-image')[0]
+        const canvas = document.createElement('canvas')
+        // Note these are scaleFactor * CreateBarcodeController.canvasWidth / canvasHeight, which is required due to scaling
+        canvas.width = 1012
+        canvas.height = 392
+        canvas.getContext('2d').drawImage(image, 0, 0, canvas.width, canvas.height)
+        canvas.toBlob(blob => {
+          navigator.clipboard.write([new ClipboardItem({ 'image/png': blob })])
+
+          const feedbackContainer = container.find('div.copy-barcode-feedback-container')
+          const feedbackContent = feedbackContainer.find('span')
+          feedbackContent.text('')
+          feedbackContainer.show()
+          feedbackContent.html('Copied barcode')
+        }, 'image/png')
+      })
+  }
+
+  const autoFocusBarcodeField = () => {
+    $('#scan-barcode-form #barcode').trigger('focus')
+  }
+
+  const extractCsrfFieldAsKeyValuePairFromForm = form => {
+    return form
+      .serializeArray()
+      .filter(formField => formField.name === '_csrf')
+      .map(formField => {
+        return `${formField.name}=${formField.value}`
+      })
+  }
+
+  const submitCookiePreferencesFormViaAjax = () => {
```

Comment:
You mightn't need all of these functions - e.g. is cookie preferences used in the staff app?

Reply:
Agreed. Removed the public facing methods which will not be used.